src/parser/internal/definition/type.rs

```diff
@@ -111,6 +103,42 @@ fn single(state: &mut State) -> ParseResult<TypeDefinition> {
 
             Ok(TypeDefinition::Dict(position, templates))
         }
+        TokenKind::Null
+        | TokenKind::True
+        | TokenKind::False
+        | TokenKind::LiteralFloat
+        | TokenKind::LiteralString
+        | TokenKind::LiteralInteger => {
+            state.iterator.next();
+
+            let value = current.value.clone();
+            let position = current.position;
+            let comments = state.iterator.comments();
+
+            let literal = match &current.kind {
+                TokenKind::Null => Literal::Null(LiteralNull { position, comments }),
+                TokenKind::True => Literal::True(LiteralTrue { position, comments }),
+                TokenKind::False => Literal::False(LiteralFalse { position, comments }),
+                TokenKind::LiteralInteger => Literal::Integer(LiteralInteger {
+                    comments,
+                    position,
+                    value,
+                }),
+                TokenKind::LiteralFloat => Literal::Float(LiteralFloat {
+                    comments,
+                    position,
+                    value,
+                }),
+                TokenKind::LiteralString => Literal::String(LiteralString {
+                    comments,
+                    position,
+                    value,
+                }),
+                _ => unreachable!(),
+            };
+
+            Ok(TypeDefinition::Literal(literal))
+        }
```

Comment:
this can be simplified like this:

```suggestion
        TokenKind::Null => Ok(TypeDefinition::Literal(Literal::Null(LiteralNull {
          comments: state.iterator.comments(),
          position,
        }))),
        TokenKind::True => Ok(TypeDefinition::Literal(Literal::True(LiteralTrue {
          comments: state.iterator.comments(),
          position,
        }))),
        TokenKind::False => Ok(TypeDefinition::Literal(Literal::False(LiteralFalse {
          comments: state.iterator.comments(),
          position,
        }))),
        TokenKind::LiteralFloat => Ok(TypeDefinition::Literal(Literal::Float(LiteralFloat {
          comments: state.iterator.comments(),
          position,
          value: current.value.clone(),
        }))),
        TokenKind::LiteralInteger => Ok(TypeDefinition::Literal(Literal::Integer(LiteralInteger {
          comments: state.iterator.comments(),
          position,
          value: current.value.clone(),
        }))),
        TokenKind::LiteralString => Ok(TypeDefinition::Literal(Literal::String(LiteralString {
          comments: state.iterator.comments(),
          position,
          value: current.value.clone(),
        }))),
```